sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -392,12 +392,16 @@ case class WholeStageCodegenExec(child: SparkPlan) extends UnaryExecNode with Co
 
     // Check if compiled code has a too large function
     if (maxCodeSize > sqlContext.conf.hugeMethodLimit) {
-      logWarning(s"Found too long generated codes and JIT optimization might not work: " +
-        s"the bytecode size was $maxCodeSize, this value went over the limit " +
+      logInfo(s"Found too long generated codes and JIT optimization might not work: " +
+        s"the bytecode size ($maxCodeSize) is above the limit " +
         s"${sqlContext.conf.hugeMethodLimit}, and the whole-stage codegen was disabled " +
         s"for this plan. To avoid this, you can raise the limit " +
-        s"${SQLConf.WHOLESTAGE_HUGE_METHOD_LIMIT.key}:\n$treeString")
-      return child.execute()
+        s"`${SQLConf.WHOLESTAGE_HUGE_METHOD_LIMIT.key}`:\n$treeString")
+      child match {
+        // For batch file source scan, we should continue executing it
+        case f: FileSourceScanExec if f.supportsBatch => // do nothing
```

Comment:
I feel a little weird `WholeStageCodegenExec` has specific error handling for each spark plan. Could we handle this error inside `FileSourceScanExec`? For example, how about checking if `parent.isInstanceOf[WholeStageCodegenExec]` in `FileSourceScanExec`?

Reply:
If we do it in `FileSourceScanExec `, we are unable to know which causes the fallback. Now, we have at least two reasons that trigger the fallback. 

Ideally, we should not call `WholeStageCodegenExec ` in `doExecute`.

Reply:
yea, I totally agree that we need to refactor this in future. Anyway, it's ok for now.